enketo/enketo-transformer
# Upgrade Node & NPM dependencies, fix NPM audit vulnerabilities

Closes #98

### Changes

- Upgraded to libxslt@0.10.0
- Pin engines to Node >=14 <17, NPM >=6 <7
- Pin Volta to Node 16/NPM 6
- Set up matrix testing for Travis CI to use NPM 14 and 16
- Upgraded [sub-]dependencies for all outstanding security vulnerabilities
- `npm update` for dependencies with semver-compatible upgrades
- Upgraded other outdated dependencies:
  - `eslint-plugin-jsdoc`
  - `mocha`
  - `xmldom`
- Upgraded Travis CI Ubuntu version to Xenial

### Validation steps taken

- `npm clean-install`
- `npm install` with previous `node_modules` in place
- `npm audit` shows 0 vulnerabilities
- `npm outdated` shows 0 packages outdated (well, it shows a false positive for `express@^5.0.0-alpha.8`)
- `npm run test` passes (this also includes `npm run eslint-check`)
- `npm run develop` + manually check a transformation loads
- temporarily `npm install grunt; npx grunt-cli transforms` before/after, with the same logic used in `enketo-core` to transform all the test fixtures, to verify they don't produce a diff
- `npm run build-docs` before/after doesn't produce a diff (because my half awake brain read "jsdoc" in `eslint-plugin-jsdoc` and instinctively thought `jsdoc`, but perhaps useful as a sanity check)
